src/metadata/mwdb_uploads.py

```diff
@@ -0,0 +1,61 @@
+import re
+import urllib.parse
+
+from metadata import Metadata
+from mwdblib import Malwarecage
+from typing import List, Any
+
+
+class MalwarecageUploadsMetadata(Metadata):
+    """
+    :param mwdb_api_token: API key for 'mquery' user in Malwarecage
+    :param mwdb_api_url: API URL accessible from mquery daemon
+    :param mwdb_url: Malwarecage URL accessible for mquery users
+    """
+    __depends_on__: List[Any] = []
+
+    def __init__(self, mwdb_api_token: str, mwdb_api_url: str, mwdb_url: str) -> None:
+        super().__init__()
+        self.mwdb = Malwarecage(api_url=mwdb_api_url, api_key=mwdb_api_token)
+        self.mwdb_url = mwdb_url
+
+    def extract(self, matched_fname, current_meta):
+        # '/uploads' Malwarecage directory format
+        # /mnt/samples/9/d/c/5/9dc571ae13a62954155999cae9cecc4f0689e2ba9a8940f81d1e564271507a3e
+        m = re.search(r"/[a-f0-9]/[a-f0-9]/[a-f0-9]/[a-f0-9]/([a-f0-9]+)$", matched_fname)
```

Comment:
Depends on how strict you want to be, regex backreference can be used here since, as it seems, the four directories inside "samples" are the first four characters of the checksum. Thus, something like this can be strctier:

```suggestion
        # /mnt/samples/9/d/c/5/9dc571ae13a62954155999cae9cecc4f0689e2ba9a8940f81d1e564271507a3e
        m = re.search(r"/([a-f0-9])/([a-f0-9])/([a-f0-9])/([a-f0-9])/\1\2\3\4([a-f0-9]+)$", matched_fname)
```

See this regex101 for example -> https://regex101.com/r/6G9xUC/2

Reply:
:+1: makes sense

Reply:
Changed to:
```python
r"/([a-f0-9])/([a-f0-9])/([a-f0-9])/([a-f0-9])/(\1\2\3\4[a-f0-9]+)$",
```
(whole basename is capturing group)